### ISSUE TITLE
In PHP 7.2, an exception is thrown if the column name is a string instead of an array when building an IN condition.

### ISSUE THREAD

**Issue by vistart** [yiisoft/yii2#13405]
### What steps will reproduce the problem?

As of PHP 7.2, `count()` only receives an array or a class that implements Countable, and throws an exception when it encounters another type.
We look at line 1186 ([2.0.10](https://github.com/yiisoft/yii2/2.0.10/master/framework/db/QueryBuilder.php#L1186)) or line 1188 ([master](https://github.com/yiisoft/yii2/blob/master/framework/db/QueryBuilder.php#L1188)), if `$column` is a string at this time, then an exception will be thrown.
As we all know, in future versions of PHP, parameter type restrictions will become more stringent.

### My Solution:

Check whether the `$column` is an array before counting it.
for example:
```php
if ($column instanceof \Traversable || (is_array($column) && count($column) > 1)) {
    return $this->buildCompositeInCondition($operator, $column, $values, $params);
} elseif (is_array($column)) {
    $column = reset($column);
}
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10 & 2.0.11-dev
| PHP version      | 7.2.0-dev
| Operating system | Any

**Comment by samdark** [comment]
Which class are you referring to?

**Comment by vistart** [comment]
@samdark Updated.
[QueryBuilder.php](https://github.com/yiisoft/yii2/blob/master/framework/db/QueryBuilder.php#L1188)

**Comment by vistart** [comment]
like [this job](https://travis-ci.org/yiisoft/yii2/jobs/192891263):
the reason for the 3), 4), 7) ~ 20), 22) ~ 29) etc. are same as below,

**Comment by cebe** [comment]
Yeah, this is known issue, many tests are failing on PHP 7.2 because of count not supporting non-arrays anymore: https://travis-ci.org/yiisoft/yii2/jobs/192891263#L305

**Comment by vistart** [comment]
The same problem also occurs in [yii2-redis](https://github.com/yiisoft/yii2-redis):
[LuaScriptBuilder.php](https://github.com/yiisoft/yii2-redis/blob/master/LuaScriptBuilder.php#L352)

**Comment by SilverFire** [comment]
Fixed by 03d53b7
